arangodb-net-standard/CollectionApi/CollectionApiClient.cs

```diff
@@ -58,5 +58,30 @@ public async Task PutCollectionTruncateAsync(string collectionName)
             }
             await GetApiErrorException(response);
         }
+
+        public async Task<GetCollectionResponse> GetCollectionAsync(GetCollectionOptions options = null)
+        {
+            try
+            {
+                if (options != null)
+                {
+                    var response = await _transport.GetAsync(_collectionApiPath + "/" + options.CollectionName);
+
+                    if (response.IsSuccessStatusCode)
+                    {
+                        var stream = await response.Content.ReadAsStreamAsync();
+                        var collection = DeserializeJsonFromStream<GetCollectionResponse>(stream, false, true);
+                        return collection;
+                    }
+                    throw await GetApiErrorException(response);
+                }
+                throw new ApiErrorException("Collection Name is required");
+            }
+            catch (Exception ex)
+            {
+                throw new ApiErrorException(ex.Message);
```

Comment:
We don't use ApiErrorException like this. It could obscure some underlying transport exception we would actually want the details of, and want the client to handle those exceptions directly.  In general, ApiErrorException is only used to capture API error responses and not other types of exception (e.g. server is offline, you are offline, some router plays a joke on you, etc). Make sense?

Reply:
In short - I think we can remove the try/catch